### DISCUSSION TITLE
--print-config with flat config

### DISCUSSION THREAD

**Discussion by alexey-koran** [eslint/eslint#16798] [Q&A]
Hello, not found any info about it, saw that u will work with redundant cli flags in phase 5
But, does right now [--print-config](https://eslint.org/docs/latest/use/command-line-interface#--print-config) works with flat config?

**Comment by nzakas** [comment]
It should. Why don't you try it and let us know what happens? :)

**Reply by alexey-koran** [reply]
> It should. Why don't you try it and let us know what happens? :)

Because I tried, and got undefined
For sure if the command is right

![image](https://user-images.githubusercontent.com/36141306/213801651-74fdd7d0-ab6a-4c12-a747-8edd8294d64c.png)

**Reply by alexey-koran** [reply]
Or, if I'm provide a file, I'll get this

![image](https://user-images.githubusercontent.com/36141306/213806634-3d0384d9-c4d4-4898-a1d6-732614a7484b.png)

**Reply by kecrily** [reply]
@alexey-koran Do you want to create an issue about this?

**Reply by nzakas** [reply]
You need to provide a file on the command line. The `--print-config` flag returns the config object for the given file.

Not all parsers are supported as of yet, but please do open an issue and be sure to mention which parser you're using so we can take a look.

**Reply by jjangga0214** [reply]
@nzakas But it's strange.

Here's a very simple example.

**eslint.config.js**

```js
import globals from "globals";
import ts from "@typescript-eslint/eslint-plugin";
import tsParser from "@typescript-eslint/parser";
import react from "eslint-plugin-react";
import reactHooks from "eslint-plugin-react-hooks";

export default [
  {
    files: ["**/*.tsx"],
    languageOptions: {
      globals: {
        ...globals.es2021,
        ...globals.browser,
      },
      parser: tsParser,
      parserOptions: {
        project: ["./tsconfig.json"],
        ecmaFeatures: {
          jsx: true,
        },
      },
    },
    plugins: {
      react,
      "react-hooks": reactHooks,
      "@typescript-eslint": ts,
    },
    rules: {
      ...react.configs.recommended.rules,
      ...reactHooks.configs.recommended.rules,
      ...ts.configs["recommended-type-checked"].rules,
    },
  },
];
```

**The result of `eslint --print-config esint.config.js`**

```json
{
  "languageOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
    "parser": "espree@9.6.1",
    "parserOptions": {},
    "globals": {}
  },
  "plugins": [
    "@"
  ],
  "rules": {}
}
```

No rules, parserOptions, globals, etc are printed.
Neither are plugins (what is `@`?)
Maybe they are only printed for js files?

**Info**

```json
{ 
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^6.17.0",
    "@typescript-eslint/parser": "^6.17.0",
    "eslint": "^8.56.0",
    "eslint-plugin-react": "^7.33.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "globals": "^13.24.0"
  }
}
```

**Reply by jjangga0214** [reply]
I wanted this to be an issue, so forwarded to #17946

**Reply by nzakas** [reply]
When you use `--print-config`,  you need to specify the file that you want to calculate the config for. By using `--print-config eslint.config.js`, you are saying "get me the config to lint `eslint.config.js`. However, your config only has settings for `.tsx` files, so you should be using: `npx eslint --print-config foo.tsx`.

**Reply by Wander95** [reply]
Ey @nzakas Did this behavior change on the new version? I was testing it on the v8.57.0 and the output was as I expected, I was working with v9.6.0 and was getting the same  results as @alexey-koran once I downgraded it worked.

![image](https://github.com/eslint/eslint/assets/56240295/094089f5-2f45-4345-9fc8-c797ebcab7aa)
